reframe/frontend/executors/policies.py

```diff
@@ -334,8 +371,15 @@ def runcase(self, case):
     def _poll_tasks(self):
         '''Update the counts of running checks per partition.'''
         getlogger().debug('updating counts for running test cases')
-        getlogger().debug('polling %s task(s)' % len(self._running_tasks))
-        for t in self._running_tasks:
+        all_running = sum(self._running_tasks.values(), [])
+        getlogger().debug('polling %s task(s)' % len(all_running))
+        for partname, sched in self._schedulers.items():
+            getlogger().debug(f'polling {len(self._running_tasks[partname])} '
+                              f'task(s) in {partname}')
+            jobs = [task.check.job for task in self._running_tasks[partname]]
+            sched.poll_jobs(jobs)
```

Comment:
I split this because I didn't fit. I will rewrite like this:
```
sched.poll_jobs(
    task.check.job for task in self._running_tasks[partname]
)
```

Reply:
The difference is that in the first case you create a list, in the second just a generator.